checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessVisitor.java

```diff
@@ -24,6 +28,9 @@ public void processClassTree(ClassTree classTree) {
         // and InitializationChecker, this checker performs the flow analysis
         // (which is handled in the BaseTypeVisitor), but does not perform
         // any type checking.
-        // Thus, this method does nothing.
+        // Thus, this method does nothing but scan through the members.
+        if (!assumeInitialized) {
```

Comment:
@flo2702 This is the key change in this PR. Sorry for adding so many other changes while tracking this down.
I think I'll merge this PR now, to fix the exception. Can you open a new PR if you find a better fix?

Reply:
Agreed. I'll keep looking, but haven't found anything better.